indigo/dataqualityreport.py

```diff
@@ -99,6 +118,81 @@ def get_count_errors_in_priority_levels(self):
         return out
 
 
+def _check_project_data_for_source_errors(input_json):
+    source_table_entries_that_are_not_used = []
+    source_ids_referenced_that_are_not_in_sources_table = []
+    source_ids_referenced = []
+    source_ids_found = []
+
+    # ----------------- Find all Source ID's referenced in data
+    for key in TYPE_PROJECT_SOURCES_REFERENCES:
+        field_value = jsonpointer.resolve_pointer(input_json, key, default="")
+        if isinstance(field_value, str):
+            for source_id in [s.strip() for s in field_value.strip().split(",")]:
+                if source_id:
+                    source_ids_referenced.append({"source_id": source_id})
+
+    for config in TYPE_PROJECT_SOURCES_REFERENCES_LIST:
+        data_list = jsonpointer.resolve_pointer(
+            input_json, config["list_key"], default=None
+        )
+        if isinstance(data_list, list) and data_list:
+            for item in data_list:
+                field_value = jsonpointer.resolve_pointer(
+                    item, config["item_source_ids_key"], default=""
+                )
+                if field_value:
+                    for source_id in [
+                        s.strip() for s in field_value.strip().split(",")
+                    ]:
+                        if source_id:
+                            source_ids_referenced.append({"source_id": source_id})
+
+    # -------------------- Now look through source table
+    source_list = jsonpointer.resolve_pointer(
+        input_json, TYPE_PROJECT_SOURCE_LIST["list_key"], default=None
+    )
+    if isinstance(source_list, list) and source_list:
+        # Look through source table looking for problems
+        for source_item in source_list:
+            source_id = jsonpointer.resolve_pointer(
+                source_item, TYPE_PROJECT_SOURCE_LIST["item_id_key"], default=""
+            )
+            if source_id:
+                source_id = str(source_id).strip()
+                source_ids_found.append(source_id)
+                # Is this source ID used? Add to list if not
+                if (
```

Comment:
Is there a reason this is formatted like this?

Reply:
black?

Reply:
Yeah, I wondered that, some funky line splits.